nucleus/track.py

```diff
@@ -0,0 +1,166 @@
+import json
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+import requests
+
+from nucleus.annotation import Annotation
+from nucleus.prediction import Prediction
+
+from .constants import (
+    ANNOTATIONS_KEY,
+    DATASET_ID_KEY,
+    METADATA_KEY,
+    OVERWRITE_KEY,
+    PREDICTIONS_KEY,
+    REFERENCE_ID_KEY,
+    SCENE_REFERENCE_ID_KEY,
+)
+
+if TYPE_CHECKING:
+    from . import NucleusClient
+
+
+@dataclass  # pylint: disable=R0902
+class Track:  # pylint: disable=R0902
+    """A track is a class of objects (annotation or prediction) that forms a one-to-many relationship
+    with objects, wherein an object is an instance of a track.
+
+    Args:
+        reference_id (str): A user-specified name of the track that describes the class of objects it represents.
+        scene_reference_id (Optional[str]): A user-specified reference ID for the scene this track belongs to.
+        metadata: Arbitrary key/value dictionary of info to attach to this track.
+    """
+
+    _client: "NucleusClient"
+    reference_id: str
+    dataset_id: str
+    scene_reference_id: Optional[str] = None
+    metadata: Optional[dict] = None
+
+    @classmethod
+    def from_json(cls, payload: dict, client: "NucleusClient"):
+        """Instantiates track object from schematized JSON dict payload."""
+        return cls(
+            _client=client,
+            reference_id=str(payload[REFERENCE_ID_KEY]),
+            dataset_id=str(payload[DATASET_ID_KEY]),
+            scene_reference_id=payload.get(SCENE_REFERENCE_ID_KEY, None),
+            metadata=payload.get(METADATA_KEY, None),
+        )
+
+    def to_payload(self) -> dict:
+        """Serializes track object to schematized JSON dict."""
+        payload: Dict[str, Any] = {
+            REFERENCE_ID_KEY: self.reference_id,
+            DATASET_ID_KEY: self.dataset_id,
+            SCENE_REFERENCE_ID_KEY: self.scene_reference_id,
+            METADATA_KEY: self.metadata,
+        }
+
+        return payload
+
+    def to_json(self) -> str:
+        """Serializes track object to schematized JSON string."""
+        return json.dumps(self.to_payload(), allow_nan=False)
+
+    def update(
```

Comment:
Knowing our customers we probably need to support mutability for metadata 🙈 Or do you think we should rather go the immutability route here and make people update by deleting and replacing the track?